text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+  requirements.
+- In addition, the infrastructure team must approve the integration of the
+  target into CI, and the CI-related requirements. This review and approval
+  will typically take place in the PR adding the target to CI.
+- In addition, the release team must approve the long-term viability of the
+  target, and the additional work of supporting the target.
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. A tier 1 target may
+  be demoted or removed if it becomes obsolete or no longer meets this
+  requirement.
+- The target must build and pass tests reliably in CI.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+- If running the testsuite requires additional infrastructure (such as systems
+  running the target), the target development team shall arrange to provide
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
+- Tier 1 targets must provide documentation for the Rust community for how to
+  build and run tests for the platform, using emulation if possible, or
+  dedicated hardware if necessary.
+- A tier 1 target may be demoted or removed if it no longer meets these
+  requirements. Any proposal for demotion or removal will be communicated
+  widely to the Rust community, both when initially proposed and before being
+  dropped from a stable release.
+- All tier 2 requirements apply.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+While these criteria attempt to document the policy, that policy still involves
+human judgment. Targets must fulfill the spirit of the requirements as well, as
+determined by the judgment of the approving teams.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The set of approving teams for each tier arose out of discussion with the
+various teams involved with aspects of the Rust project impacted by new
+targets.
+
+
+# Prior art
+[prior-art]: #prior-art
+
+This attempts to formalize and document Rust policy around targets and
+architectures. https://forge.rust-lang.org/release/platform-support.html
+documents some of that policy.
+
+Future expansions of such policy may find requirements from other communities
+useful as examples, such as Debian's [arch
+policy](https://release.debian.org/bullseye/arch_policy.html) and [archive
+criteria](https://ftp-master.debian.org/archive-criteria.html).
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Where, precisely, should we post the official policy?
+- How should we track the maintainers of a target, so that we can page them if
+  we need an issue addressed involving that target? Can we do so using github
+  teams, so that we can mention them in an issue or PR to get their attention?
```

Comment:
The drawback of GitHub membership is the "Member" badge, which might give external contributors' comments an "official" appearance. Instead we could add them (like icebreakers-llvm) as a "marker team" in the team repo, and ping them with:

```
@rustbot ping team-name
```

Reply:
I would hesitate on adding a whole bunch of teams without being sure that it's worth the overhead. At least initially I think putting a table at the end of this document containing the platforms and the people who maintain them, might be enough for most cases, and we can add teams for platforms that have a large enough group of maintainers.

Reply:
> at the end of this document

I guarantee that this will quickly go out of scope / never be updated.

If we wish to put the onus on people to opt-in, we have have magic comments in source code files that we grep for, or use something like [GitHub's code owners](https://github.blog/2017-07-06-introducing-code-owners/)

Reply:
I've added some discussion of this. Marker teams seem like a good idea. Perhaps we could additionally add support in rustbot for automatically pinging a marker team when an issue or PR is labeled with a specific label? Would that be feasible?